#!/usr/bin/env python
PACKAGE = "spare_function"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# name - a string which specifies the name under which this parameter should be stored
# type - defines the type of value stored, and can be any of int_t, double_t, str_t, or bool_t
# level - A bitmask which will later be passed to the dynamic reconfigure callback. When the callback is called all of the level values for parameters that have been changed are ORed together and the resulting value is passed to the callback.
# description - string which describes the parameter
# default - specifies the default value
# min - specifies the min value (optional and does not apply to strings and bools)
# max - specifies the max value (optional and does not apply to strings and bools)

gen.add("set_point",   bool_t,   0, "set_point",  False)

gen.add("target_yaw", double_t, 0, "target_yaw",    0,  -3.14,   3.14)

gen.add("target_x", double_t, 0, "target_x",    0,  -100,   100)
gen.add("target_y", double_t, 0, "target_y",    0,    -100,   100)

gen.add("max_speed", double_t, 0, "max_speed",    0.2,    0,   10)
gen.add("min_speed", double_t, 0, "min_speed",    -0.2,    -10,   10)


exit(gen.generate(PACKAGE, "spare_function", "spare_function_"))

